query-engine/metrics/Cargo.toml

```diff
@@ -14,6 +16,7 @@ tracing.workspace = true
 tracing-futures = "0.2"
 tracing-subscriber = "0.3.11"
 parking_lot = "0.12"
+pin-project = "1"
```

Comment:
Could `pin-project` become a `workspace` dependency? We use it in `driver-adapters` already.